src/usePopper.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+import { useLayoutEffect, useRef, useState, useMemo } from 'react';
+import {
+  createPopper as defaultCreatePopper,
+  type Options as PopperOptions,
+} from '@popperjs/core';
+
+type Options = { ...PopperOptions, createPopper: typeof defaultCreatePopper };
+
+type State = {
+  styles: {
+    [key: string]: $Shape<CSSStyleDeclaration>,
+  },
+  attributes: {
+    [key: string]: { [key: string]: string | boolean },
+  },
+};
+
+export const usePopper = (
+  referenceElement: Element,
+  popperElement: HTMLElement,
+  options: Options = {}
+) => {
+  const popperInstanceRef = useRef();
+  const [state, setState] = useState<State>({
+    styles: {
+      popper: { position: 'absolute', left: '0', top: '0' },
+    },
+    attributes: {},
+  });
+  const createPopper = useMemo(
+    () => options.createPopper || defaultCreatePopper,
+    [options.createPopper]
+  );
+  const popperOptions = useMemo(() => {
+    // eslint-disable-next-line no-unused-vars
+    const { createPopper, ...popperOptions } = options;
+    return popperOptions;
+  }, [options]);
+
+  const updateStateModifier = useMemo(
+    () => ({
+      name: 'updateState',
+      enabled: true,
+      phase: 'write',
+      fn: ({ state }) => {
+        const elements = Object.keys(state.elements);
+
+        setState({
+          styles: Object.fromEntries(
+            elements.map((element) => [element, state.styles[element] || {}])
+          ),
+          attributes: Object.fromEntries(
+            elements.map((element) => [element, state.attributes[element]])
+          ),
+        });
+      },
+      requires: ['computeStyles'],
+    }),
+    [setState]
+  );
+
+  useLayoutEffect(() => {
```

Comment:
`useLayoutEffect` needs to check browser env check to avoid the SSR warning

Reply:
🤦‍♂️

Reply:
pinging this again

Reply:
I'm wondering if we could follow the React team advice where they suggest not to run the code at all on server-side?